libs/cardano-cddl/app/Main.hs

```diff
@@ -0,0 +1,34 @@
+{-# LANGUAGE ImportQualifiedPost #-}
+
+module Main where
+
+import Codec.CBOR.Cuddle.CDDL (sortCDDL)
+import Codec.CBOR.Cuddle.Huddle (Huddle, toCDDL)
+import Codec.CBOR.Cuddle.Pretty ()
+import Data.Foldable (for_)
+import Prettyprinter (Pretty (pretty))
+import Prettyprinter.Render.Text (hPutDoc)
+import System.IO (IOMode (..), hPutStrLn, withFile)
+import Test.Cardano.Ledger.Conway.CDDL qualified as Conway
+
+-- | Message to be appended to the top of every generated file
+preface :: String
+preface =
+  "; This file was auto-generated by cuddle-cddl. Please do not modify it directly!"
+
+-- | Map from spec to the CDDL file location
+cuddleSpecs :: [(Huddle, String)]
+cuddleSpecs =
+  [ (Conway.conway, "./eras/conway/impl/cddl-files/conway-merged.cddl")
+  ]
+
+writeSpec :: Huddle -> String -> IO ()
+writeSpec hddl path =
+  let cddl = toCDDL hddl
+   in withFile path WriteMode $ \h -> do
+        hPutStrLn h preface
+        hPutDoc h (pretty $ sortCDDL cddl)
```

Comment:
Why do we sort CDDL? I would really like the order of definition from the original Huddle file to be preserved.
```suggestion
        hPutDoc h (pretty cddl)
```

Messing up the order prevents us form logically grouping related types together

Reply:
Do you know a way to do this? Currently, the only effect of this sorting is to have a single alphabetical list, rather than a list that has first the "rules" and then the "groups" (in CDDL these aren't distinguished, but it's helpful to do so in Huddle).

I guess perhaps using template Haskell might allow finding the actual order in the Huddle file?

Reply:
I see ... So if we don't sort the huddle we'll first get this order first

```haskell
[block, transaction]
      <> [kes_signature, language, potential_languages, signkeyKES]
```

followed or interleaved by all other reusable definitions.

Some sort of template haskell approach would probably work. However, it looks tricky, so unless you can easily figure out how to do it, we probably shouldn't worry about it at this point. In other words, if it looks like it'll be too much work then just create an issue on the cuddle repo about it and we can look into it at some point later time permitting

Reply:
https://github.com/input-output-hk/cuddle/issues/35